### DISCUSSION TITLE
Would bmake features be elligible for inclusion if people chose to implement them?

### DISCUSSION THREAD

**Discussion by anoraktrend** [rocky/remake#154] [Q&A] [answered]
It'd be nice to have a single make to rule them all for BSDs, Linux, and other Unix-likes. Would pull requests implementing these features be outright rejected? Would they be considered beyond the scope of this project?

**Comment by rocky** [comment, marked as answer]
Tentatively, I this kind of thing is fine. Since semantics may be different between GNU and BSD, I imagine here would need to be some way turn on and off different behaviors such as by testing an environment variable setting.

Also in the past in other projects, it has sometimes happens that for a small change a lot of code has to be disrupted beyond recognition, and beyond the ability to easily understand and maintain the code.  However as long as things are not pathological like this, it is fine to do this here.

**Reply by anoraktrend** [reply]
To summarize and clarify for myself and others to understand:
"Yes, so long as the divergent behaviors can be changed via environment variable and the changes don't impact the ability to easily understand and maintain the code. " 

Would that be an accurate summary? I don't want to change what you mean by attempting understand.

**Reply by rocky** [reply]
Yes, perfect.
